examples/01-tensor.rs

```diff
@@ -1,48 +1,51 @@
 //! Intro to dfdx::arrays and dfdx::tensor
 
-use dfdx::{
-    shapes::{Const, HasShape, Rank1, Rank2, Rank3},
-    tensor::{AsArray, Cpu, OnesTensor, SampleTensor, Tensor, TensorFrom, ZerosTensor},
-};
```

Comment:
I actually agree with you on that matter. The reason I used `prelude` nonetheless is, that it automatically imports the Cuda device if the according feature is enabled. Otherwise the `dfdx::tensor::cuda::device::Cuda` struct has to imported with a `#[cfg(feature = "cuda")]` safe guard. 

I can change that if desired but I think it would look "ugly":

```rust
#[cfg(feature = "cuda")]
use dfdx::tensor::cuda::device::Cuda;
#[cfg(feature = "cuda")]
type Device = Cuda;
```

That could well just be me though.

Reply:
Oh yeah great point.

Since we are doing type aliases for the devices we could not include cpu/cuda in use statements, and then do:
```rust
#[cfg(feature = "cuda")]
type Device = dfdx::tensor::Cuda;
#[cfg(not(feature = "cuda"))]
type Device = dfdx::tensor::Cpu;
```

Probably not as clean as doing prelude, but I think better than doing conditional use

Reply:
Done now :)